src/gui/folder.cpp

```diff
@@ -1664,6 +1666,41 @@ void Folder::slotAboutToRemoveAllFiles(SyncFileItem::Direction dir, std::functio
     msgBox->open();
 }
 
+void Folder::slotNeedToRemoveRemnantsReadOnlyFolders(const QList<SyncFileItemPtr> &folders,
+                                                     const QString &localPath,
+                                                     std::function<void (bool)> callback)
+{
+    auto listOfFolders = QStringList{};
+    for (const auto &oneFolder : folders) {
+        listOfFolders.push_back(oneFolder->_file);
+    }
+
+    const auto msg = tr("Do you want to clean up remnant read-only folders left over from previous failed synchronization attempts.\n%1");
+    auto msgBox = new QMessageBox(QMessageBox::Question, tr("Remove remnant invalid folders?"),
+                                  msg.arg(listOfFolders.join("\n")), QMessageBox::NoButton);
```

Comment:
I know that for us this makes sense but my feeling is that a normal user will have absolutely no idea what this information means, I think we may need to provide a better explanation of what the issue is and why cleaning up is a good idea